docs/explanation/security.md

```diff
@@ -0,0 +1,63 @@
+
+Your charm repository should have tooling configured so that any dependencies with security updates are detected automatically (such as [Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/about-dependabot-security-updates) or [Renovate](https://www.mend.io/renovate/)), prompting to you re-lock so that the charm will be built with the latest version.
+
+For information about supported versions and how to report security issues, please see [SECURITY.md](https://github.com/canonical/operator/blob/main/SECURITY.md).
+
+## Risks
+
+Using Ops does not introduce any new security risks to charms, unless the charm is integrated with a tracing receiver. It does expand the impact of the Juju risk of an attacker gaining access to the filesystem of the charm:
+
+* Access to the deferred notice queue provides information about events that could not be immediately processed (this includes the event name and the event context provided in the hook environment at the time).
+* Access to trace data provides detailed information about the implementation of the charm and the events that it has processed. This is particularly the case when tracing has not been configured, as the charm will have a large amount of buffered trace data stored (when tracing is active, this will be regularly sent to the trace receiver and removed from the local database).
+
+Ops inherits all the risks of Juju executing charms (for example, injecting data into the charm context or through the use of hook commands). Charm authors should be familiar with {external+juju:ref}`Juju security <juju-security>` and {external+pebble:ref}` Pebble security <security>`.
+
+If a charm is integrated with a tracing receiver, this introduces the risk of outgoing traces being intercepted. Traces should not include any sensitive data, but intercepted traces can provide information about the structure of the charm and the events that the charm has processed, and blocking trace data can hide malicious activity.
+
+## Good practices
+
+* Never include any sensitive data in logs.
+* Never include any sensitive data in traces.
+* Use {external+juju:ref}`Juju secrets <secret>` for storing and sharing sensitive data.
+* Juju users that integrate a charm with a tracing receiver should also integrate with a certificate provider, to ensure all traces are sent via HTTPS.
+* Charms should follow best practices for writing secure Python code.
+* Charms should have workflows that statically check for security issues (such as [ruff](https://docs.astral.sh/ruff/linter/), [bandit](https://bandit.readthedocs.io/en/latest/index.html), and [zizmor](https://docs.zizmor.sh/)).
```

Comment:
Hmmm I dunno if I would recommend bandit.

Reply:
Because you'd recommend just enabling the flake8-bandit rules in ruff instead? Or for some other reason? I'd use ruff these days, but bandit's pretty commonly used across Canonical charms and it doesn't seem unreasonable to me to continue to use it if you haven't adopted ruff.